src/swiss_dict.jl

```diff
@@ -0,0 +1,653 @@
+    GC.@preserve slots begin
+        return unsafe_load(convert(Ptr{UInt8}, pointer(slots)), i)
+    end
+end
+
+Base.@propagate_inbounds function _slotset!(slots::Vector{_u8x16},  v::UInt8, i::Int)
+   @boundscheck 0 < i <= length(slots)*16 || throw(BoundsError(slots, 1 + (i-1)>>4))
+    GC.@preserve slots begin
+        return unsafe_store!(convert(Ptr{UInt8}, pointer(slots)), v, i)
+    end
+end
+
+@inline function _find_candidates(v::_u8x16, tag::UInt8)
+    match = _vcmp_eq(v, _expand16(tag))
+    return (match, v[16].value === 0x00)
+end
+
+@inline _find_free(v::_u8x16) = _vcmp_le(v, _expand16(UInt8(1)))
+
+# Basic operations
+
+# get the index where a key is stored, or -1 if not present
+ht_keyindex(h::SwissDict, key) = ht_keyindex(h::SwissDict, key, _hashtag(hash(key))...)
+function ht_keyindex(h::SwissDict, key, i0, tag)
+    slots = h.slots
+    keys = h.keys
+    sz = length(slots)
+    i = i0 & (sz-1)
+    _prefetchr(pointer(h.keys, i*16+1))
+    _prefetchr(pointer(h.vals, i*16+1))
+    #Todo/discuss: _prefetchr(pointer(h.keys, i*16+9))?
+    @inbounds while true
+        msk = slots[i+1]
+        cands, done = _find_candidates(msk, tag)
+        while cands != 0
+            off = trailing_zeros(cands)
+            idx = i*16 + off + 1
+            isequal(keys[idx], key) && return idx
+            cands = _blsr(cands)
+        end
+        done && break
+        i = (i+1) & (sz-1)
+    end
+    return -1
+end
+
+# get the index where a key is stored, or -pos if not present
+# and the key would be inserted at pos
+# This version is for use by setindex! and get!. It never rehashes.
+ht_keyindex2!(h::SwissDict, key) = ht_keyindex2!(h, key, _hashtag(hash(key))...)
+@inline function ht_keyindex2!(h::SwissDict, key, i0, tag)
+    slots = h.slots
+    keys = h.keys
+    sz = length(slots)
+    i = i0 & (sz-1)
+    _prefetchw(pointer(h.keys, i*16+1))
+    _prefetchw(pointer(h.vals, i*16+1))
+    #Todo/discuss: _prefetchr(pointer(h.keys, i*16+9))?
+    @inbounds while true
+        msk = slots[i+1]
+        cands, done = _find_candidates(msk, tag)
+        while cands != 0
+            off = trailing_zeros(cands)
+            idx = i*16 + off + 1
+            isequal(keys[idx], key) && return idx, tag
+            cands = _blsr(cands)
+        end
+        done && break
+        i = (i+1) & (sz-1)
+    end
+    i = i0 & (sz-1)
+    @inbounds while true
+        msk = slots[i+1]
+        cands = _find_free(msk)
+        if cands != 0
+            off = trailing_zeros(cands)
+            idx = i*16 + off + 1
+            return -idx, tag
+        end
+        i = (i+1) & (sz-1)
+    end
+end
+
+function _setindex!(h::SwissDict, v, key, index, tag)
+    @inbounds h.keys[index] = key
+    @inbounds h.vals[index] = v
+    h.count += 1
+    h.age += 1
+    so =  _slotget(h.slots, index)
+    h.nbfull += (iszero(index & 0x0f) & (so==0x00))
+    _slotset!(h.slots, tag, index)
+    if index < h.idxfloor
+        h.idxfloor = index
```

Comment:
do we need to fix the tests to make sure to cover this?

Reply:
I can't seem to find a test to cover this. It's quite hard to find a test case specifically for this. But, this serves the purpose of `h.idxfloor`, so it isn't wrong.